src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -486,14 +480,7 @@ static void notify_execbuf_xgq(struct kds_command *xcmd, int status)
 		client_stat_inc(client, scu_c_cnt[xcmd->cu_idx]);
 	}
 
-	if (status == KDS_COMPLETED)
-		ecmd->state = ERT_CMD_STATE_COMPLETED;
-	else if (status == KDS_ERROR)
-		ecmd->state = ERT_CMD_STATE_ERROR;
-	else if (status == KDS_TIMEOUT)
-		ecmd->state = ERT_CMD_STATE_TIMEOUT;
-	else if (status == KDS_ABORT)
-		ecmd->state = ERT_CMD_STATE_ABORT;
+	ecmd->state = kds_ert_table[status];
```

Comment:
See above, duplicate code, status should be an enumeration, and it should be checked.